go/service/saltpack.go

```diff
@@ -245,20 +224,139 @@ func (h *SaltpackHandler) SaltpackVerifyString(ctx context.Context, arg keybase1
 		Source: strings.NewReader(arg.SignedMsg),
 	}
 
-	spui := &capSaltpackUI{}
-	uis := libkb.UIs{
-		IdentifyUI: h.NewRemoteIdentifyUI(arg.SessionID, h.G()),
-		SecretUI:   &nopSecretUI{},
-		SaltpackUI: spui,
-		SessionID:  arg.SessionID,
-	}
-	eng := engine.NewSaltpackVerify(h.G(), earg)
-	m := libkb.NewMetaContext(ctx, h.G()).WithUIs(uis)
-	if err := engine.RunEngine2(m, eng); err != nil {
+	spui, err := h.frontendVerify(ctx, arg.SessionID, earg)
+	if err != nil {
 		return keybase1.SaltpackVerifyResult{}, err
 	}
 	res := keybase1.SaltpackVerifyResult{
 		Plaintext: sink.String(),
+		Verified:  spui.verified,
+	}
+	if spui.signingKID != nil {
+		res.SigningKID = *spui.signingKID
+	}
+	if spui.sender != nil {
+		res.Sender = *spui.sender
+	}
+	return res, nil
+}
+
+func (h *SaltpackHandler) SaltpackEncryptFile(ctx context.Context, arg keybase1.SaltpackEncryptFileArg) (string, error) {
```

Comment:
there's a not trivial amount of logic in all of this plumbing, and it seems to me like it's important that `Encrypt`/`Decrypt` and `Sign`/`Verify` remain symmetric. i think it's worth writing a couple of tests that do a super simple in-and-out for both of these flows. we don't need to retest saltpack, but if it's not too bad to set up two users with keys and blah blah blah, i think it's worth knowing this file won't break. it'll also make it easier to add support for teams in the future.

Reply:
tests added